src/targeting.js

```diff
@@ -31,16 +31,25 @@ export let filters = {
 
 // If two bids are found for same adUnitCode, we will use the highest one to take part in auction
 // This can happen in case of concurrent auctions
-export function getHighestCpmBidsFromBidPool(bidsReceived, highestCpmCallback) {
+// If adUnitBidLimit is set above 0 return top N number of bids
+export function getHighestCpmBidsFromBidPool(bidsReceived, highestCpmCallback, adUnitBidLimit = 0) {
   const bids = [];
   // bucket by adUnitcode
   let buckets = groupBy(bidsReceived, 'adUnitCode');
   // filter top bid for each bucket by bidder
   Object.keys(buckets).forEach(bucketKey => {
+    let bucketBids = [];
     let bidsByBidder = groupBy(buckets[bucketKey], 'bidderCode');
-    Object.keys(bidsByBidder).forEach(key => bids.push(bidsByBidder[key].reduce(highestCpmCallback)));
+    Object.keys(bidsByBidder).forEach(key => bucketBids.push(bidsByBidder[key].reduce(highestCpmCallback)));
+    // if adUnitBidLimit is set, pass top N number bids
+    if (adUnitBidLimit > 0) {
+      bucketBids.sort((a, b) => b.cpm - a.cpm);
```

Comment:
I'd like to see the bid list randomized before sorting so that `aardvark` doesn't always win ties.

Reply:
I've been told that the order of the list is chronological by time of bid response. This is ok.